Add "shadowable" check; rename jf

ryanlaws opened the pull request:
Add get_shadowable_attribute, which checks whether the track is a
shadow, removing some duplication. Use this for is_jf, which has been
renamed. This could be used for other shadowable attributes.

ryanlaws committed:
Add "shadowable" check; rename jf

Add get_shadowable_attribute, which checks whether the track is a
shadow, removing some duplication. Use this for is_jf, which has been
renamed. This could be used for other shadowable attributes.

ryanlaws committed:
Fix backwards ternary

ryanlaws committed:
Leave commands alone!

tyleretters commented on lib/Track.lua:
```diff
@@ -42,7 +42,7 @@ function Track:new(x)
   -- crow
   t.pair = 1
-  t.jf = false
+  t.is_jf = false
```
the `is_` prefix is reserved for methods. the attributes are always whatever follows the underscore.

tyleretters replied:
please change back to `t.jf`

ryanlaws replied:
Changed back and modified `get_shadowable_attribute` check - see 2c04cd3

tyleretters commented on lib/Track.lua:
```diff
@@ -274,10 +274,18 @@ function Track:get_shadow_attribute(attribute)
   elseif attribute == "device"  then return track:get_device()
   elseif attribute == "pair"    then return track:get_pair()
-  elseif attribute == "jf"      then return track:is_jf()
+  elseif attribute == "is_jf"   then return track:get_is_jf()
```
similar to the `t.jf` comment - `is_` is the convention for boolean methods. the `get_` is not used with boolean getters.

ryanlaws replied:
Changed back and modified get_shadowable_attribute check - see 2c04cd3

tyleretters commented on lib/Track.lua:
```diff
@@ -274,10 +274,18 @@ function Track:get_shadow_attribute(attribute)
   end
 end
 
+function Track:get_shadowable_attribute(attribute)
```
this is good. i'll need to test but i think this will help with some of the issues.

ryanlaws replied:
Sounds good!

ryanlaws committed:
Un-rename jf things, check is_ in shadowable

Convention is to prefix Boolean getters with is_. Revert to use that, and
modify shadowable attribute check to use it

ryanlaws committed:
Un-rename set_jf

tyleretters commented:
nice!!!
